### ISSUE TITLE
cue/load: using qualifier that is the same as final path element causes import failure

### ISSUE THREAD

**Issue by rogpeppe** [cue-lang/cue#3162]
### What version of CUE are you using (`cue version`)?

<pre>
a90aec5c611bda3d73dfe484d4794e07499ff2ae
</pre>

### Does this issue reproduce with the latest stable release?

Yes (at v0.8.2)

### What did you do?

```
# old
env CUE_EXPERIMENT=modules=0
exec cue vet -c .

# new
env CUE_EXPERIMENT=modules
exec cue vet -c .

-- cue.mod/module.cue --
module: "github.com/foo/bar"
language: {
	version: "v0.9.0"
}
-- bar.cue --
package bar

import xxx "github.com/foo/bar/abc:abc"

z: xxx.x & 123

-- abc/baz.cue --
package abc

x: 123
```

### What did you expect to see?

A passing test

### What did you see instead?

```
# old (0.023s)
# new (0.011s)
> env CUE_EXPERIMENT=modules
> exec cue vet -c .
[stderr]
github.com/foo/bar@v0: import failed: no dependency found for package "github.com/foo/bar/abc":
    ./bar.cue:3:8
[exit status 1]
FAIL: /tmp/testscript3893947838/repro.txtar/script.txtar:7: unexpected command failure
```

**Comment by myitcv** [comment]
It was ambiguous which vet call had failed, so I've updated the original description. 

@rogpeppe - to confirm we will fix this as part of the work in #3155?

**Comment by rogpeppe** [comment]
> It was ambiguous which vet call had failed, so I've updated the original description.
> 
> @rogpeppe - to confirm we will fix this as part of the work in #3155?

The fix is actually independent of that, but I have a fix and it will land at or around the same time.
